### PR TITLE
issue1 - create models methods

### PR DESCRIPTION
**Pull request by vsseixaso**
Implementation of the methods of the models. Organization of code and translation of model comments.

### CODE REVIEW

**Comment by joseims** [comment]
Troca o nome de weapon-generator pra item-generator

File: weapon-generator.cpp [hunk 1]
```diff
@@ -1,13 +1,14 @@
 #include <bits/stdc++.h>
-using namespace std;
```
Comment (joseims) [inline weapon-generator.cpp]: Muda o nome pra item-generator

**Review by joseims — CHANGES_REQUESTED** [review]

**Review by raonismaneoto — CHANGES_REQUESTED** [review]
Translate the comments and insert some description in the pr about what you've done

File: alterer-generator.cpp [hunk 2]
```diff
@@ -1,45 +1,36 @@
...
 struct alterer {
     string text;
     int strength;
     int defense;
     int health;
 };
 
-//defautl
-//Existe tanto para monstros quanto para armas
+// defautl: there is as much for monsters as for weapons and armor
```
Comment (raonismaneoto) [inline alterer-generator.cpp]: default*

File: alterer-generator.cpp [hunk 3]
```diff
@@ -48,15 +39,38 @@ alterer hairy;
...
+alterer imaginary;
+alterer legless;
+
+alterer monster_alterer[13] = {pacifist, dumb, blind, imaginary, legless,
     invisible, gigantic, three_headed, terrifying, hairy, stony, murderer, default_};
 
-alterer get_random_item_alterer(){//Pega um alterador aleatório da lista
-    int random = rand() % item_alterer.size();
+alterer get_random_item_alterer() {
+    int random = rand() % 9;
```
Comment (raonismaneoto) [inline alterer-generator.cpp]: extract this number to a well named constant

File: alterer-generator.cpp [hunk 4]
```diff
@@ -48,15 +39,38 @@ alterer hairy;
...
     return item_alterer[random];
 }
 
-alterer get_random_monster_alterer(){//Pega um alterador aleatório da lista
-    int random = rand() % monster_alterer.size();
+alterer get_random_monster_alterer() {
+    int random = rand() % 13;
```
Comment (raonismaneoto) [inline alterer-generator.cpp]: same thing

File: alterer-generator.cpp [hunk 5]
```diff
@@ -48,15 +39,38 @@ alterer hairy;
...
     return monster_alterer[random];
+}
+
+void generate_generic_alterer(alterer &a) {
+    a.text = "text";
+    a.strength = 10;
```
Comment (raonismaneoto) [inline alterer-generator.cpp]: where these numbers come from? You gotta wrap them in well named constants

File: item-generator.cpp [hunk 6]
```diff
@@ -65,21 +61,39 @@ item join_item_alterer(item i, alterer a, int gameProgressMultiplyer) {
     return new_item;
 }
 
-item generate_weapon(int gameProgressMultiplyer) {//Gera uma arma aleatória
+item generate_weapon(int gameProgressMultiplyer) {
```
Comment (raonismaneoto) [inline item-generator.cpp]: follow the language pattern: game_progress_multiplier*

**Review by raonismaneoto — CHANGES_REQUESTED** [review]

File: alterer-generator.cpp (hunk 5 above)
Comment (vsseixaso) [reply alterer-generator.cpp]: this is a random number for temporary testing until we have defined the final values. On the new commit I specified

File: item-generator.cpp [hunk 7]
```diff
@@ -29,64 +29,66 @@ item heavy_armor;// +def
...
 
 item get_random_armor() {
-    int random = rand() % 5;
+    int random = rand() % N_ARMORS;
     return armors[random];
 }
 
-int multiplyer_item(int attribute, int gameProgressMultiplyer) {
-    return attribute * gameProgressMultiplyer;
+int multiplyer_item(int attribute, int game_progress_multiplayer) {
```
Comment (raonismaneoto) [inline item-generator.cpp]: multiplier*

**Review by raonismaneoto — CHANGES_REQUESTED** [review]

**Review by raonismaneoto — APPROVED** [review]

File: item-generator.cpp [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    i.strength = 10 + (rand() % 5);
+    i.defense = 10 + (rand() % 5);
+    i.health = 10 + (rand() % 5);
+    i.attack = "attack";
+    i.type = "type";
+    i.price = 10 + (rand() % 5);
+}
+
+void generate_all_items() {
+    for (int i = 0; i < 4; i++) {
```
Comment (joseims) [inline item-generator.cpp]: Change the numbers for N__ARMORS and N_WEAPONS

File: monster-generator.cpp [hunk 9]
```diff
@@ -22,9 +22,49 @@ monster vampire;
...
+    m.defense = 10 + (rand() % 5);
+    m.health = 10 + (rand() % 5);
+    m.attack = "attack";
+    m.progress_multiplier = 10 + (rand() % 5);
+    m.coins = 10 + (rand() % 5);
+}
 
-void generate_random_monster(bool isBoss, int gameProgressMultiplyer) {
+void generate_all_monsters() {
+    for (int i = 0; i < 7; i++) {
```
Comment (joseims) [inline monster-generator.cpp]: 7 for N_monsters

**Review by joseims — CHANGES_REQUESTED** [review]

**Review by joseims — APPROVED** [review]

**Pull request merged**